MediaManager/Plugin.MediaManager.Abstractions/Implementations/MediaManagerBase.cs

```diff
@@ -255,7 +291,8 @@ private void OnPlayingChanged(object sender, PlayingChangedEventArgs e)
         {
             if (sender != CurrentPlaybackManager) return;
             MediaFinished?.Invoke(sender, e);
-            await PlayNext();
+            if (MediaQueue.Repeat == RepeatType.RepeatOne) await Seek(TimeSpan.Zero);
```

Comment:
This should be solved in the PlayNext() method, where the system should also take advantage of if a song has been added twice to the list.